card/mod/01_core/set/all/rules.rb

```diff
@@ -194,53 +200,48 @@ def all_user_ids_with_rule_for set_card, setting_code
     end
   end
 
-  def user_rule_cards user_name, setting_code
+  def preference_cards user_name, setting_code
     Card.search(
       { right: { codename: setting_code },
         left: { left: { type_id: SetID }, right: user_name }
-        }, "rule cards for user: #{user_name}"
+      }, "rule cards for user: #{user_name}"
     )
   end
 
   def rule_cache
     Card.cache.read('RULES') || begin
-      rule_hash = {}
-      all_rule_keys_with_id do |key,rule_id|
-        rule_hash[key] = rule_id
-      end
+      @rule_hash = {}
+      @user_ids_hash = {}
+      @rule_keys_hash = {}
 
-      user_ids_hash = {}
-      rule_keys_hash = {}
-      all_user_rule_keys_with_id_and_user_id do |key, rule_id, user_id|
-        rule_hash[ user_rule_key(key,user_id) ] = rule_id
-        user_ids_hash[key] ||= []
-        user_ids_hash[key] << user_id
-        rule_keys_hash[user_id] ||= []
-        rule_keys_hash[user_id] << key
-      end
-      write_user_ids_cache user_ids_hash
-      write_rule_keys_cache rule_keys_hash
-      write_rule_cache rule_hash
+      interpret_simple_rules
+      interpret_preferences
+
+      write_user_ids_cache @user_ids_hash
+      write_rule_keys_cache @rule_keys_hash
+      write_rule_cache @rule_hash
     end
   end
 
-  def user_rule_key key, user_id
+  def preference_key key, user_id
     "#{key}+#{user_id}"
   end
 
   # all users that have a user-specific rule for a given rule key
   def user_ids_cache
     Card.cache.read('USER_IDS') || begin
+      clear_rule_cache
```

Comment:
good sign if two people make the same change independently :) I added the same line to my phases PR a few minutes ago to fix failing notification cukes.